src/AppInstallerCLICore/Workflows/PortableFlow.cpp

```diff
@@ -281,15 +297,124 @@ namespace AppInstaller::CLI::Workflow
+                const auto& expectedHash = uninstallEntry[PortableValueName::SHA256];
+                std::string expectedHashValue = expectedHash.has_value() ? uninstallEntry[PortableValueName::SHA256].value().GetValue<Value::Type::String>() : "";
+
+                if (std::filesystem::exists(targetPathValue))
+                {
+                    std::ifstream inStream{ targetPathValue, std::ifstream::binary };
+                    const Utility::SHA256::HashBuffer& targetFileHash = SHA256::ComputeHash(inStream);
+                    inStream.close();
+
+                    bool overrideHashMismatch = context.Args.Contains(Execution::Args::Type::HashOverride);
+
+                    if (!SHA256::AreEqual(SHA256::ConvertToBytes(expectedHashValue), targetFileHash))
+                    {
+                        if (overrideHashMismatch)
+                        {
+                            context.Reporter.Warn() << Resource::String::PortableHashMismatchOverridden << std::endl;
+                        }
+                        else
+                        {
+                            context.Reporter.Warn() << Resource::String::PortableHashMismatchOverrideRequired << std::endl;
+                            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_PORTABLE_UNINSTALL_FAILED);
+                        }
+                    }
+
+                    std::filesystem::remove(targetPathValue);
+                    AICLI_LOG(CLI, Info, << "Successfully deleted portable exe:" << targetPathValue);
+                }
+                else
+                {
+                    AICLI_LOG(CLI, Info, << "Portable exe not found; Unable to delete portable exe: " << targetPathValue);
+                }
+            }
+            else
+            {
+                AICLI_LOG(CLI, Info, << "The registry value for [TargetFullPath] does not exist");
+            }
+        }
+
+        void RemoveInstallDirectory(Execution::Context& context)
+        {
+            PortableARPEntry& uninstallEntry = context.Get<Execution::Data::PortableARPEntry>();
+            const auto& installDirectory = uninstallEntry[PortableValueName::InstallLocation];
+
+            if (installDirectory.has_value())
+            {
+                const std::filesystem::path& installDirectoryValue = installDirectory.value().GetValue<Value::Type::UTF16String>();
+                bool isUpdate = WI_IsFlagSet(context.GetFlags(), Execution::ContextFlag::InstallerExecutionUseUpdate);
+
+                if (context.Args.Contains(Execution::Args::Type::Purge) ||
+                    (!isUpdate && Settings::User().Get<Settings::Setting::UninstallPurgePortablePackage>() && !context.Args.Contains(Execution::Args::Type::Preserve)))
+                {
+                    context.Reporter.Warn() << Resource::String::PurgeInstallDirectory << std::endl;
+                    const auto& removedFilesCount = std::filesystem::remove_all(installDirectoryValue);
```

Comment:
I'm not sure that we should support purging a directory that we did not create.

Reply:
Changed the flow around so that purging only works if we created the directory. Shows warning message to user if purge is provided and we did not create the directory.